pkg/util/timeutil/time_zone_util.go

```diff
@@ -66,3 +67,45 @@ func ParseFixedOffsetTimeZone(location string) (offset int, origRepr string, suc
 	}
 	return offset, strings.TrimSuffix(strings.TrimPrefix(origRepr, "("), ")"), true
 }
+
+// TimeZoneOffsetStringConversion converts a time string to offset seconds
+// Supported time zone strings :- GMT/UTC±[00:00:00 - 15:59:59], GMT/UTC±[0-16]
+// Unsupported time zone strings :- GMT/UTC±16:00 to upper, GMT/UTC±6.5
+// (case insensitive)
+func TimeZoneOffsetStringConversion(s string) (offset int64, ok bool) {
+	pattern := `(?i)(GMT|UTC)[+-]((((0?\d)|(1[0-5])):([0-5]\d))|(((0?\d)|(1[0-5])):([0-5]\d):([0-5]\d))|((1+[0-6])|(0?\d)))\b$`
```

Comment:
this pattern is getting a little complicated :( -- we're doing a lot of offset checking using strings whereas i think it'd be cleaner to do it by comparing the `offset` against the min/max.

i think it may be cleaner to do something like
`(?i)(GMT|UTC)([+-])(\d{1,3}(:[0-5]\d){0,2})` instead (allow up to 1-3 digits first, and there can be between 0 and 2 digit components), and we can check more out of bounds conditions by comparing `abs(offsetSecs)` to `167*60*60 + 59*60`

Reply:
Thank you for simplifying the regex. I agree this is a better approach